src/ExportHistory/Models/ExportJobCreationOptions.cs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Microsoft.DurableTask.Client;
+
+namespace Microsoft.DurableTask.ExportHistory;
+
+/// <summary>
+/// Configuration for a export job.
+/// </summary>
+public record ExportJobCreationOptions
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ExportJobCreationOptions"/> class.
+    /// </summary>
+    public ExportJobCreationOptions()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ExportJobCreationOptions"/> class.
+    /// </summary>
+    /// <param name="mode">The export mode (Batch or Continuous).</param>
+    /// <param name="completedTimeFrom">The start time for the export based on completion time (inclusive). Required for Batch mode. For Continuous mode, this will be set to UtcNow if not provided.</param>
+    /// <param name="completedTimeTo">The end time for the export based on completion time (inclusive). Required for Batch mode, null for Continuous mode.</param>
+    /// <param name="destination">The export destination where exported data will be stored. Required unless default storage is configured.</param>
+    /// <param name="jobId">The unique identifier for the export job. If not provided, a GUID will be generated.</param>
+    /// <param name="format">The export format settings. Optional, defaults to jsonl-gzip.</param>
+    /// <param name="runtimeStatus">The orchestration runtime statuses to filter by. Optional.</param>
+    /// <param name="maxInstancesPerBatch">The maximum number of instances to fetch per batch. Optional, defaults to 100.</param>
+    /// <exception cref="ArgumentException">Thrown when validation fails.</exception>
+    public ExportJobCreationOptions(
+        ExportMode mode,
+        DateTimeOffset completedTimeFrom,
+        DateTimeOffset? completedTimeTo,
+        ExportDestination? destination,
+        string? jobId = null,
+        ExportFormat? format = null,
+        List<OrchestrationRuntimeStatus>? runtimeStatus = null,
+        int? maxInstancesPerBatch = null)
+    {
+        // Generate GUID if jobId not provided
+        this.JobId = string.IsNullOrEmpty(jobId) ? Guid.NewGuid().ToString("N") : jobId;
+
+        if (mode == ExportMode.Batch)
+        {
+            if (completedTimeFrom == default)
+            {
+                throw new ArgumentException(
+                    "CompletedTimeFrom is required for Batch export mode.",
+                    nameof(completedTimeFrom));
+            }
+
+            if (!completedTimeTo.HasValue)
```

Comment:
Any reason we do a different check here than `== default`? Not that I'm sure what the `default` means there, but it's important to be consistent for readability.